grunttasks/generate-valid-styles.js

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var path = require('path');
+var stripSpecialChars = require('./strip-special-chars');
```

Comment:
Having a non-task js file in with the rest of the tasks is a bit confusing, maybe put this under a `lib` subdirectory.